src/components/TreeGrid.ts

```diff
@@ -46,31 +48,30 @@ const wrapContent =
     state: 'closed',
   });
 
-export const hierarchyMapToRows = <T extends RowContents>(hierarchyMap: Map<T, T[]>): Row<T>[] => {
-  const traverseHierarchy = (parent: T, depth: number, previousRows: Row<T>[]): Row<T>[] => {
+export const populateTreeFromRoot = <T extends RowContents>(root: T): Row<T>[] => {
+  const createRows = (parent: RowContents, depth: number, previousRows: Row<T>[]): Row<T>[] => {
```

Comment:
How come we can't type `parent: T`?

Reply:
I was puzzled by this too. I think the problem is that `child` is coming from `parent.children` which is directly typed as `RowContents`, not `T`. So `child` is type `RowContents` but `parent` is type `T` which may be different.

I did try to fix this by making `RowContents` itself parameterized but couldn't get it to work.

Hmm, I could move the `as T` from line 56 below to the recursive call to `creeateRows()`, which might be a little better.